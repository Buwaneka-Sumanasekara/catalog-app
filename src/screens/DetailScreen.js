import React from 'react';
import { useRoute } from '@react-navigation/native';


const DetailScreen = () => {
  const {params} = useRoute()

  const itemId=params?.itemId || ""


  

  return  <DefaultWrapper>
      
  </DefaultWrapper>
};

export default DetailScreen;
