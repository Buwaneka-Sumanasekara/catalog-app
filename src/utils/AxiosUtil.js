import axios from 'axios';

export const getAxios = () => {
  const axiosInstance = axios.create({
    baseURL: process.env.API_URL,
    timeout: process.env.DEFAULT_TIMEOUT ? parseInt(process.env.DEFAULT_TIMEOUT) : 0,
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
  });

  return axiosInstance;
};
