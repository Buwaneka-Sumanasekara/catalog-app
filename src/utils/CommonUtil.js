import Globals from '../constants/Globals';
import ScreenNames from '../constants/ScreenNames';

function removeItemFromArray(ar, item, key = '') {
  return ar.filter((v) => v[key] !== item[key]);
}

function getAnimeStatusByScreenKey(screenKey) {
  let status = Globals.AnimeStatus.Airing;
  switch (screenKey) {
    case ScreenNames.HomeCompleteScreen.key:
      status = Globals.AnimeStatus.Complete;
      break;
    case ScreenNames.HomeUpComingScreen.key:
      status = Globals.AnimeStatus.Upcoming;
      break;
    default:
      status = Globals.AnimeStatus.Airing;
      break;
  }
  return status;
}

const checkValueExistInArray = (value, ar = [], key = '') => {
  const found = ar.find((item) => value[key] === item[key]);
  return !!found || false;
};

function getPaddings(columns, padding, index) {
  let paddingLeft = padding / 2;
  let paddingRight = padding / 2;
  let paddingTop = padding / 2;
  const paddingBottom = padding / 2;

  const row = Math.floor(index / columns);
  const isFirstItem = index % columns === 0 || index % columns === columns;
  const isLastItem = index % columns === columns - 1;
  if (row === 0) {
    //first row
    paddingTop = padding;
    if (isFirstItem) {
      paddingLeft = 0;
    } else if (isLastItem) {
      paddingRight = 0;
    }
  } else {
    if (isFirstItem) {
      paddingLeft = 0;
    } else if (isLastItem) {
      paddingRight = 0;
    }
  }

  return { paddingLeft, paddingRight, paddingTop, paddingBottom };
}

export { removeItemFromArray, getAnimeStatusByScreenKey, checkValueExistInArray, getPaddings };
