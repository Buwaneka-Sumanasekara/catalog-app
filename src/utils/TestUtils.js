import React from 'react';
import { render } from '@testing-library/react-native';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { QueryClient, QueryClientProvider } from 'react-query';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import { NavigationContainer } from '@react-navigation/native';

const middlewares = [thunk];
const store = configureStore(middlewares);
const initialState = {
  favorites: {
    items: [],
    count: 0,
  },
};

const queryClient = new QueryClient();

const AllTheProviders = ({ children }) => {
  return (
    <Provider store={store(initialState)}>
      <QueryClientProvider client={queryClient}>
        <SafeAreaProvider>
          <NavigationContainer>{children}</NavigationContainer>
        </SafeAreaProvider>
      </QueryClientProvider>
    </Provider>
  );
};

const customRender = (children) => render(<AllTheProviders>{children}</AllTheProviders>);

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render };
