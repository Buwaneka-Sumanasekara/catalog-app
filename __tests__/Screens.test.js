import DetailScreen from '../src/screens/DetailScreen';
import { render } from '../src/utils/TestUtils'

describe('All Screens renders snapshots check:', () => {
  it('DetailScreen', () => {
      const { toJSON } = render(<DetailScreen/>)
      expect(toJSON()).toMatchSnapshot();
  })
})
