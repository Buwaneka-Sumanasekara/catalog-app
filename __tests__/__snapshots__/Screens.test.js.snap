// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All Screens renders snapshots check: DetailScreen 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#ffffff",
        "flex": 1,
        "paddingHorizontal": 16,
      },
      undefined,
      {
        "paddingHorizontal": 0,
        "paddingVertical": 0,
      },
    ]
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          accessibilityRole="progressbar"
          accessibilityState={
            {
              "busy": true,
            }
          }
          accessible={true}
          style={
            [
              {
                "alignItems": "center",
                "justifyContent": "center",
              },
              {
                "marginBottom": 20,
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              {
                "height": 48,
                "opacity": 1,
                "width": 48,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "alignItems": "center",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "height": 48,
                    "transform": [
                      {
                        "rotate": "45deg",
                      },
                    ],
                    "width": 48,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "height": 24,
                      "overflow": "hidden",
                      "width": 48,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "height": 48,
                        "transform": [
                          {
                            "translateY": 0,
                          },
                          {
                            "rotate": "-165deg",
                          },
                        ],
                        "width": 48,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        {
                          "height": 24,
                          "overflow": "hidden",
                          "width": 48,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          {
                            "borderColor": "rgba(103, 80, 164, 1)",
                            "borderRadius": 24,
                            "borderWidth": 4.8,
                            "height": 48,
                            "width": 48,
                          }
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              collapsable={false}
              style={
                {
                  "alignItems": "center",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "height": 48,
                    "transform": [
                      {
                        "rotate": "45deg",
                      },
                    ],
                    "width": 48,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "height": 24,
                      "overflow": "hidden",
                      "top": 24,
                      "width": 48,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "height": 48,
                        "transform": [
                          {
                            "translateY": -24,
                          },
                          {
                            "rotate": "345deg",
                          },
                        ],
                        "width": 48,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        {
                          "height": 24,
                          "overflow": "hidden",
                          "width": 48,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          {
                            "borderColor": "rgba(103, 80, 164, 1)",
                            "borderRadius": 24,
                            "borderWidth": 4.8,
                            "height": 48,
                            "width": 48,
                          }
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
                "letterSpacing": 0,
              },
              {
                "writingDirection": "ltr",
              },
              undefined,
            ]
          }
        >
          Hold on...
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
